compiler/src/dotty/tools/dotc/transform/ContextFunctionResults.scala

```diff
@@ -101,8 +101,9 @@ object ContextFunctionResults:
   def contextFunctionResultTypeAfter(meth: Symbol, depth: Int)(using Context) =
     def recur(tp: Type, n: Int): Type =
       if n == 0 then tp
-      else tp match
+      else tp.dealias match
         case defn.FunctionTypeOfMethod(mt) => recur(mt.resType, n - 1)
+        case _ => tp
```

Comment:
`tp.dealias` on its own is enough to fix the issue.

Do we want to crash if we don't match `defn.FunctionTypeOfMethod(mt)`? If yes, we shouldn't add this line.

Reply:
Alright, default case has been removed.